### PR TITLE
Cortex secret management

### PR DESCRIPTION
**Pull request by infa-bapa**
## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: charts/cortex-agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -89,10 +89,12 @@ spec:
 
         - name: agent-ids
           mountPath: /etc/traps
-
+        
+        {{- if .Values.agent.createSecret }}
```
Comment (ranberant) [inline charts/cortex-agent/templates/daemonset.yaml:93]: We need the same `if` statement in `deployment-secret.yaml` file in order to not create the `Secret`

File: charts/cortex-agent/values.yaml [hunk 2]
```diff
@@ -31,6 +31,9 @@ agent:
   # Agent distribution id should be retrieved from the tenant
   distributionId: ""
 
+  # Set to "false" to inject distributionId externally, e.g. Vault-agent-injector
+  createSecret: true
```
Comment (ranberant) [inline charts/cortex-agent/values.yaml:35]: Let's create a new hierarchy outside of `agent` (which contains agent specific settings that aren't related to k8s), like this:
```
deploymentSecret:
  create: true
```

Comment (infa-bapa) [reply charts/cortex-agent/values.yaml]: Added as below

```
daemonsetSecret:
  create: true
```

File: charts/cortex-agent/templates/daemonset.yaml (hunk 1 above)
Comment (infa-bapa) [reply charts/cortex-agent/templates/daemonset.yaml]: Added the ``` if ``` statement in ```deployment-secret.yaml``` as well

**Comment by ranberant** [comment]
@infa-bapa Thanks for the Pull Request. Due to changes in policy i will have to push it from my branch though.

**Comment by infa-ddeore** [comment]
@ranberant  can you also release this helm chart please?

```
$ helm template cortex-xdr --repo https://paloaltonetworks.github.io/cortex-helm --version 1.1.0
Error: chart "cortex-xdr" version "1.1.0" not found in https://paloaltonetworks.github.io/cortex-helm repository
```

**Pull request closed without merging**